src/main/scala/org/graphframes/GraphFrame.scala

```diff
@@ -466,8 +465,15 @@ class GraphFrame private(
       indexedVertices.select(
         col(ATTR + "." + ID).cast("long").as(LONG_ID), col(ATTR + "." + ID).as(ID), col(ATTR))
     } else {
-      val indexedVertices = zipWithUniqueId(vertices)
-      indexedVertices.select(col("uniq_id").as(LONG_ID), col("row." + ID).as(ID), col("row").as(ATTR))
+      val withLongIds = vertices.select(ID)
+        .distinct()
```

Comment:
Yes, but it doesn't specify which method it uses and it doesn't allow users to set a partition number. Based on the optimized plan, it does shuffle twice. I just wanted to put a quick fix for @nchammas to test whether this is the root cause.